arcaflow_plugin_horreum_client/horreum_client_plugin.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+import sys
+import typing
+import requests
+import json
+
+from arcaflow_plugin_sdk import plugin
+from horreum_client_schema import InputParams, SuccessOutput, ErrorOutput
+
+
+@plugin.step(
+    id="horreum-client",
+    name="Horreum Client",
+    description="Uploads an object to the Horreum server",
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def horreum_client(
+    params: InputParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+    auth_url = (
+        params.horreum_keycloak_url + "/realms/horreum/protocol/openid-connect/token"
+    )
+    auth_obj = {
+        "username": params.horreum_username,
+        "password": params.horreum_password,
+        "grant_type": "password",
+        "client_id": "horreum-ui",
+    }
+
+    try:
+        print("==> Authenticating with keycloak...")
+        auth_return = requests.post(auth_url, data=auth_obj, verify=False)
```

Comment:
Why are you hard-coding `verify=False` (especially when addressing the source of your security!)?

If you need to be able to disable TLS verification, the switch should be external to the code.

Ditto line 56.

Reply:
Good catch. I've made this an input option now.